monkestation/code/modules/blood_datum/components/advert_force_speak.dm

```diff
@@ -0,0 +1,134 @@
+	var/implanted = syndi || HAS_TRAIT(source, TRAIT_SPONSOR_IMPLANT)
+	var/bad_speak = !source.can_speak() || HAS_TRAIT(source, TRAIT_ANXIOUS) || HAS_TRAIT(source, TRAIT_SOFTSPOKEN)
+	if (!source.can_speak_language(/datum/language/common))
+		bad_speak = TRUE
+	// else if (HAS_TRAIT(source, TRAIT_SIGN_LANG))
+	// 	bad_speak = FALSE
+
+	var/list/ad_list = syndi ? GLOB.advertisements.syndi_ads : GLOB.advertisements.nt_ads
+	var/ad_idx = rand(1, ad_list.len / 2) * 2
+	var/sponsor = ad_list[ad_idx - 1]
+	var/message = ad_list[ad_idx]
+	var/spans = syndi ? list("red", "syndi-propaganda") : list()
+
+	message = replacetext(message, "%name%", source.real_name || source.name)
+
+	if(implanted && bad_speak)
+		speaker_implant_say(source, message, sponsor)
+	else
+		source.say(message, language=/datum/language/common, forced="sponsorship ([sponsor])", spans=spans)
+
+	if(implanted)
+		var/datum/bank_account/bank_account = source.get_bank_account()
+		if(bank_account)
+			var/revenue = syndi ? -40 : 5
+			bank_account.adjust_money(revenue, "[sponsor]: Sponsorship Payment")
+			bank_account.bank_card_talk("Sponsorship Payment: [revenue] credits received from [sponsor].")
+
+/datum/component/advert_force_speak/proc/speaker_implant_say(mob/living/source, message, sponsor, spans)
+	if(speaker_implant == null)
+		speaker_implant = new()
+		speaker_implant.loc = source
+		speaker_implant.owner = source
+		speaker_implant.bubble_icon = "machine"
+
+	speaker_implant.body_maptext_height_offset = source.body_maptext_height_offset
+	speaker_implant.say(message, language=/datum/language/common, forced="sponsorship ([sponsor])", spans=spans)
+
+/mob/living/speaker_implant
+	name = "speaker implant"
+	var/mob/living/owner
+
+/mob/living/speaker_implant/GetVoice()
+	if(owner == src)
+		return ""
+	return owner.GetVoice()
+
+/mob/living/speaker_implant/get_alt_name()
+	return "'s Implant"
+
+/mob/living/speaker_implant/say(message, bubble_type, list/spans = list(), sanitize = TRUE, datum/language/language = null, ignore_spam = FALSE, forced = null, filterproof = null, message_range = 7, datum/saymode/saymode = null)
+	. = ..()
+	if(.)
+		var/ending = copytext_char(message, -1)
+		var/sound/speak_sound
+		if(ending == "?")
+			speak_sound = voice_type2sound[voice_type]["?"]
+		else if(ending == "!")
+			speak_sound = voice_type2sound[voice_type]["!"]
+		else
+			speak_sound = voice_type2sound[voice_type][voice_type]
+		playsound(src, speak_sound, 300, 1, SHORT_RANGE_SOUND_EXTRARANGE-2, falloff_exponent = 0, pressure_affected = FALSE, ignore_walls = FALSE, use_reverb = FALSE, mixer_channel = CHANNEL_MOB_SOUNDS)
+
+GLOBAL_DATUM_INIT(advertisements, /datum/advertisements, new)
+
+/datum/advertisements
+	var/list/nt_ads
+	var/list/syndi_ads
+
+/datum/advertisements/New()
+	nt_ads = load_file("monkestation/strings/advertisements.txt")
+	syndi_ads = load_file("monkestation/strings/advertisements_syndi.txt")
+
+/datum/advertisements/proc/load_file(file_name)
```

Comment:
why is this some bespoke format instead of like, TOML or JSON?

Reply:
I wanted a simple format for comfortably writing many lines of text without having to mess with the specifics of the syntax
I can switch to JSON or TOML if necessary but I think this is much more user friendly, especially for non-coders who want to add their own lines